### PR TITLE
[Merged by Bors] - feat: port Counterexamples.SorgenfreyLine

### PR DESCRIPTION
**Pull request by urkud**
---
One more playground for porting archive/counterexamples

- [x] depends on: #5078

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: scripts/lint-style.sh [hunk 1]
```diff
@@ -38,7 +38,7 @@ set -exo pipefail
 touch scripts/style-exceptions.txt
 
 # Assuming `archive/` and `counterexamples/` are ported too, they need to be added here.
-find Mathlib -name '*.lean' | xargs ./scripts/lint-style.py
+git ls-files '*.lean' | grep -v '^test/' | xargs ./scripts/lint-style.py
```
Comment (eric-wieser) [inline scripts/lint-style.sh:41]:
```suggestion
find Mathlib Counterexamples Archive -name '*.lean' | xargs ./scripts/lint-style.py
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -76,15 +76,19 @@ jobs:
...
+          git ls-files 'MathlibExtras/*.lean' | LC_ALL=C sort | sed 's/\.lean//;s,/,.,g;s/^/import /' > MathlibExtras.lean
 
       - name: update Mathlib/Tactic.lean
         run: |
-          find Mathlib/Tactic -name "*.lean" | LC_ALL=C sort | sed 's/\.lean//;s,/,.,g;s/^/import /' > Mathlib/Tactic.lean
+          git ls-files 'Mathlib/Tactic/*.lean' | LC_ALL=C sort | sed 's/\.lean//;s,/,.,g;s/^/import /' > Mathlib/Tactic.lean
+
+      - name: update Counterexamples.lean
+        run: |
+          git ls-files 'Counterexamples/*.lean' | LC_ALL=C sort | sed 's/\.lean//;s,/,.,g;s/^/import /' > Mathlib/Counterexamples.lean
```
[the remark this answers is not in the document]
Comment (urkud) [reply .github/workflows/build.yml:91]: I don't know. I copied `MathlibExtras` setup.

File: scripts/lint-style.sh (hunk 1 above)
Comment (urkud) [reply scripts/lint-style.sh:41]: IMHO, `git ls-files` is better for people like me who want to run the script locally and may have files not added to `git`.

File: .github/workflows/build.yml.in [hunk 3]
```diff
@@ -56,15 +56,23 @@ jobs:
...
-          find Mathlib/Tactic -name "*.lean" | LC_ALL=C sort | sed 's/\.lean//;s,/,.,g;s/^/import /' > Mathlib/Tactic.lean
+          git ls-files 'Mathlib/Tactic/*.lean' | LC_ALL=C sort | sed 's/\.lean//;s,/,.,g;s/^/import /' > Mathlib/Tactic.lean
+
+      - name: update Counterexamples.lean
+        run: |
+          git ls-files 'Counterexamples/*.lean' | LC_ALL=C sort | sed 's/\.lean//;s,/,.,g;s/^/import /' > Counterexamples.lean
+
+      - name: update Archive.lean
+        run: |
+          git ls-files 'Archive/*.lean' | LC_ALL=C sort | sed 's/\.lean//;s,/,.,g;s/^/import /' > Archive.lean
```
Comment (eric-wieser) [inline .github/workflows/build.yml.in:75]: It looks like you didn't regenerate the other yml files

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#5078
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#5078~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**